### DISCUSSION TITLE
Why not use custom attribute rather than class?

### DISCUSSION THREAD

**Discussion by J-Cake** [tailwindlabs/tailwindcss#1488] [General]
Just a quick thought, In order to specify a list of utilities on an element, why not use a custom attribute instead of classes, they are just as easy to use in CSS (maybe they require a few more characters but that's why there's JS under the hood, right?) 

So for example if I wanted an element with a 2nd degree border radius, I would specify it like this:

```
<div class="rounded">Hi</div>
```

If I now know that I need to select that element in JS later, I can add another class or ID to it.

```
<div class="rounded sample-div">Hi</div>
```

But as is the nature of Tailwind, the classlist quickly gets long. Is it possible for me to compile TailWind to allow me to use a different attribute?

```
<div class="sample-div" tailwind="rounded">Hi</div>
```

The only thing that would stop is the ability to use JS's `HTMLElement.classList`. But TailWind can create a custom function to extend the `HTMLElement` with a function to add tailwind classes

```HTMLElement.tailwindStyle```

It would be used in the same way the `classList` would be.

Just curious. Thanks for any input

**Comment by PositiveUser** [comment]
> TailWind can create a custom function to extend the `HTMLElement` with a function to add tailwind classes

Do not change native JS object.

**Reply by J-Cake** [reply]
If Tailwind is installed and available, what's the issue with it?

**Reply by PositiveUser** [reply]
[_“The native prototypes should never be extended unless it is for the sake of compatibility with newer JavaScript features.”_](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain#:~:text=the%20native%20prototypes%20should%20never%20be%20extended%20unless%20it%20is%20for%20the%20sake%20of%20compatibility%20with%20newer%20JavaScript%20features.)

_- MDN_

**Comment by merchako** [comment]
Even if the community thinks this is a bad idea, I'm thankful that @J-Cake posted it because I had the same question. Especially since I'm adding Tailwind to a pre-existing project, it would have been useful to mentally separate traditional classes from Tailwind classes.

The Tailwind plugin for [Prettier will sort the non-Tailwind classes to the front](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier#how-classes-are-sorted) of the class string, making them easier to identify. 


```diff 
- <div class="p-3 shadow-xl select2-dropdown">
+ <div class="select2-dropdown p-3 shadow-xl">
  <!-- ... -->
</div>
```
> Any custom classes that don’t come from Tailwind plugins (like classes for targeting a third-party library) are always sorted to the front, so it’s easy to see when an element is using them:
